tests/cbmc/sources/make_common_datastructures.c

```diff
@@ -74,93 +76,112 @@ const char *nondet_c_str_is_allocated(size_t max_size)
 {
     struct s2n_dh_params *dh_params = malloc(sizeof(*dh_params));
-    if (dh_params != NULL) {
-        dh_params->dh = DH_new();
-    }
+    cbmc_populate_s2n_dh_params(dh_params);
     return dh_params;
 }
 
 EVP_MD_CTX* cbmc_allocate_EVP_MD_CTX() {
     EVP_MD_CTX *ctx = malloc(sizeof(*ctx));
-    if (ctx != NULL) {
-        ctx->digest = malloc(sizeof(*(ctx->digest)));
-        ctx->md_data = malloc(EVP_MAX_MD_SIZE);
-        ctx->pctx = malloc(sizeof(*(ctx->pctx)));
-        if (ctx->pctx != NULL) {
-            ctx->pctx->pkey = malloc(sizeof(*(ctx->pctx->pkey)));
-            if (ctx->pctx->pkey != NULL) {
-                ctx->pctx->pkey->ec_key = malloc(sizeof(*(ctx->pctx->pkey->ec_key)));
-                if (ctx->pctx->pkey->ec_key != NULL) {
-                    ctx->pctx->pkey->ec_key->group = malloc(sizeof(*(ctx->pctx->pkey->ec_key->group)));
-                    if (ctx->pctx->pkey->ec_key->group != NULL) {
-                        ctx->pctx->pkey->ec_key->group->order = malloc(sizeof(*(ctx->pctx->pkey->ec_key->group->order)));
-                        if (ctx->pctx->pkey->ec_key->group->order != NULL) {
-                            ctx->pctx->pkey->ec_key->group->order->d = malloc(sizeof(*(ctx->pctx->pkey->ec_key->group->order->d)));
-                        }
-                    }
-                    ctx->pctx->pkey->ec_key->priv_key = malloc(sizeof(*(ctx->pctx->pkey->ec_key->priv_key)));
-                    if (ctx->pctx->pkey->ec_key->priv_key != NULL) {
-                        ctx->pctx->pkey->ec_key->priv_key->d = malloc(sizeof(*(ctx->pctx->pkey->ec_key->priv_key->d)));
+    CBMC_ENSURE_REF(ctx);
+    ctx->digest  = malloc(sizeof(*(ctx->digest)));
+    ctx->md_data = malloc(EVP_MAX_MD_SIZE);
+    ctx->pctx    = malloc(sizeof(*(ctx->pctx)));
+    if (ctx->pctx != NULL) {
+        ctx->pctx->pkey = malloc(sizeof(*(ctx->pctx->pkey)));
+        if (ctx->pctx->pkey != NULL) {
+            ctx->pctx->pkey->ec_key = malloc(sizeof(*(ctx->pctx->pkey->ec_key)));
+            if (ctx->pctx->pkey->ec_key != NULL) {
+                ctx->pctx->pkey->ec_key->group = malloc(sizeof(*(ctx->pctx->pkey->ec_key->group)));
+                if (ctx->pctx->pkey->ec_key->group != NULL) {
+                    ctx->pctx->pkey->ec_key->group->order = malloc(sizeof(*(ctx->pctx->pkey->ec_key->group->order)));
+                    if (ctx->pctx->pkey->ec_key->group->order != NULL) {
+                        ctx->pctx->pkey->ec_key->group->order->d =
+                            malloc(sizeof(*(ctx->pctx->pkey->ec_key->group->order->d)));
                     }
                 }
+                ctx->pctx->pkey->ec_key->priv_key = malloc(sizeof(*(ctx->pctx->pkey->ec_key->priv_key)));
+                if (ctx->pctx->pkey->ec_key->priv_key != NULL) {
+                    ctx->pctx->pkey->ec_key->priv_key->d = malloc(sizeof(*(ctx->pctx->pkey->ec_key->priv_key->d)));
+                }
             }
         }
     }
```

Comment:
That's a lot of nesting. You're not making it any worse, so this isn't a blocker, but can we maybe make it better?

```suggestion
    CBMC_ENSURE_REF(ctx->pctx);
    ctx->pctx->pkey = malloc(sizeof(*(ctx->pctx->pkey)));
    CBMC_ENSURE_REF(ctx->pctx->pkey);
    ctx->pctx->pkey->ec_key = malloc(sizeof(*(ctx->pctx->pkey->ec_key)));
    CBMC_ENSURE_REF(ctx->pctx->pkey->ec_key);
    ctx->pctx->pkey->ec_key->group = malloc(sizeof(*(ctx->pctx->pkey->ec_key->group)));
    if (ctx->pctx->pkey->ec_key->group != NULL) {
        ctx->pctx->pkey->ec_key->group->order = malloc(sizeof(*(ctx->pctx->pkey->ec_key->group->order)));
        if (ctx->pctx->pkey->ec_key->group->order != NULL {
            ctx->pctx->pkey->ec_key->group->order->d = malloc(sizeof(*(ctx->pctx->pkey->ec_key->group->order->d)));
        }
    }
    ctx->pctx->pkey->ec_key->priv_key = malloc(sizeof(*(ctx->pctx->pkey->ec_key->priv_key)));
    if (ctx->pctx->pkey->ec_key->priv_key != NULL) {
        ctx->pctx->pkey->ec_key->priv_key->d = malloc(sizeof(*(ctx->pctx->pkey->ec_key->priv_key->d)));
    }
```